android/app/src/main/java/com/zulipmobile/notifications/FCMPushNotifications.kt

```diff
@@ -2,44 +2,50 @@
 
 package com.zulipmobile.notifications
 
+import android.annotation.SuppressLint
 import android.app.Notification
 import android.app.NotificationChannel
 import android.app.NotificationManager
 import android.app.PendingIntent
 import android.content.Context
 import android.content.Intent
 import android.graphics.Color
-import android.media.AudioAttributes
 import android.net.Uri
 import android.os.Build
 import android.os.Bundle
-import android.provider.Settings
-import android.text.TextUtils
 import android.util.Log
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.Person
+import androidx.core.graphics.drawable.IconCompat
 import com.facebook.react.ReactApplication
 import me.leolin.shortcutbadger.ShortcutBadger
 
 import com.zulipmobile.BuildConfig
 import com.zulipmobile.R
 
 private val CHANNEL_ID = "default"
-private val NOTIFICATION_ID = 435
 
 @JvmField
 val ACTION_CLEAR = "ACTION_CLEAR"
 
 @JvmField
 val EXTRA_NOTIFICATION_DATA = "data"
 
-private fun getNotificationManager(context: Context): NotificationManager {
-    return context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
-}
-
 fun createNotificationChannel(context: Context) {
     if (Build.VERSION.SDK_INT >= 26) {
         val name = context.getString(R.string.notification_channel_name)
-        val channel = NotificationChannel(CHANNEL_ID, name, NotificationManager.IMPORTANCE_HIGH)
-        getNotificationManager(context).createNotificationChannel(channel)
+
+        @SuppressLint("WrongConstant")
+        // Android Studio's linter demands NotificationManager.IMPORTANCE_* and rejects any other
+        // value, hence using "SupressLint".
+        val channel =
+            NotificationChannel(CHANNEL_ID, name, NotificationManagerCompat.IMPORTANCE_HIGH).apply {
+                description = context.getString(R.string.notification_channel_description)
```

Comment:
Hmm -- can you post a screenshot of how this description appears in the UI? That would help me wordsmith the text for the description.

I'm not actually seeing anything that looks like this when I browse around the notification settings for other apps on my phone, which is making me suspect that my phone (Android 11 on a Pixel) just ignores the description. I also don't see anything that looks like it in the handy guide doc you linked:
https://developer.android.com/training/notify-user/channels

Reply:
<img width="331" alt="Screenshot 2021-08-19 at 10 12 20 PM" src="https://user-images.githubusercontent.com/40268170/130109223-5abdf968-ed5b-4ac9-af2a-f44806983041.png">

At the very end it is the grayed out message. This is accessible on tapping the notification channel name from the app setting.

Reply:
Fascinating, thanks.

Do you see this on any other apps you happen to have lying around? I just looked at this part of the UI for several well-known apps -- including Facebook Messenger, which has an extensive set of notification categories, so they've clearly paid attention to this area of the API, and also including Android Messages, which ditto plus it's actually from the Android group at Google -- and I don't see a grayed-out message like that at all. There's the "Advanced" section, and then nothing after that. (Also, just to confirm: what Android version are you on?)

If other app makers, including the Google Android group itself, aren't using this feature, then we clearly don't need to feel obligated to do so, and I'm inclined to skip it. I think the category's name itself can convey all the same information -- perhaps we should rename it "All messages" to be clearer -- and the way the UI ends up presenting this description text feels pretty out-of-context and not likely to be helpful.

Reply:
I looked again at the settings UI where this name appears, and I think "Messages" works well as the name and "All messages" would be unnecessarily redundant.

A couple of quick comparison points:
* FB Messenger has "Chats and calls".
* Android Messages has "Incoming messages".